38-linreg-vanilla/main.go

```diff
@@ -17,27 +18,44 @@ import (
 
 func main() {
 	iterations := flag.Int("n", 1000, "number of iterations")
+	outPath := flag.String("o", "out.png", "path to output file")
 	flag.Parse()
 
-	xs, ys, err := readData("data.txt")
+	inPath := flag.Arg(0)
+	if inPath == "" {
+		// default to data.txt for downwards-compatibility
+		inPath = "data.txt"
+	}
+
+	file, err := os.Open(inPath)
```

Comment:
I'd call these `in` and `out` maybe? `file` and `f` are not very meaningful

Reply:
Yes, I renamed the file variables for clarity.